### ISSUE TITLE
Weird Validation-Loss Values

### ISSUE THREAD

**Issue by namcho** [ROCmSoftwarePlatform/tensorflow-upstream#1238]
**System information**
- Custom code: https://gist.github.com/namcho/9e161233437192815cf436361fc04435
- OS Platform and Distribution: Ubuntu 20.04.1
- TensorFlow installed from binary
- TensorFlow version: tensorflow-rocm 2.4.0
- Python version: 3.7.9
- ROCm version: 4.0
- GPU model and memory: AMD Radeon RX-580 4GB

**Describe the current behavior**
Validation-Loss values are always getting huge or NaN(from the beginning of the training to the end). Checkout below please:

> Epoch 27/40
782/782 [==============================] - 18s 23ms/step - loss: 0.9782 - accuracy: 0.8015 - val_loss: nan - val_accuracy: 0.1246
Epoch 28/40
782/782 [==============================] - 18s 23ms/step - loss: 0.9743 - accuracy: 0.7978 - val_loss: 4088319030793366629515264.0000 - val_accuracy: 0.1229
Epoch 29/40
782/782 [==============================] - 18s 23ms/step - loss: 0.9730 - accuracy: 0.7980 - val_loss: 4446401173914317259538432.0000 - val_accuracy: 0.1401
Epoch 30/40
782/782 [==============================] - 18s 23ms/step - loss: 0.9611 - accuracy: 0.8033 - val_loss: 3822467437506826088742912.0000 - val_accuracy: 0.1411
Epoch 31/40
782/782 [==============================] - 18s 23ms/step - loss: 0.9538 - accuracy: 0.8062 - val_loss: 4013053720899246293516288.0000 - val_accuracy: 0.1515

**Describe the expected behavior**
As well known validation-loss is percent value so it could not be bigger than 100 but the val-loss are getting 4013053720899246293516288.0000 kind of meaningless values.

**Standalone code to reproduce the issue**
See gist above.

**Other info / logs**
Terminal logs for cnn_cifar10.py: https://gist.github.com/namcho/d82441528a7f6082835f4772cb3e2a2d

**Comment by namcho** [comment]
Same issue also exist for validation-accuracy.

> 782/782 [==============================] - 18s 24ms/step - loss: 0.7056 - accuracy: 0.8818 - val_loss: nan - val_accuracy: 4526353220062855299072.0000
Epoch 35/40
782/782 [==============================] - 18s 24ms/step - loss: 0.6873 - accuracy: 0.8882 - val_loss: nan - val_accuracy: 2405965065804557844480.0000

I found similar issue under keras-team/keras for simple neural - network: https://github.com/keras-team/keras/issues/14215
Issue opener claimed that same code had worked well with NVIDIA GEFORCE RTX 2060(without ROCm)

**Comment by xuhuisheng** [comment]
Duplicate of https://github.com/RadeonOpenCompute/ROCm/issues/1265

**Comment by boriswinner** [comment]
This is an issue with ROCm higher than 3.5.1 with Polaris 10. The solution is to downgrade.

Here is my guide to downgrade to ROCm 3.5.1 + TensorFlow 2.2:
https://github.com/boriswinner/RX580-rocM-tensorflow-ubuntu20.4-guide

**Comment by namcho** [comment]
> This is an issue with ROCm higher than 3.5.1 with Polaris 10. The solution is to downgrade.
> 
> Here is my guide to downgrade to ROCm 3.5.1 + TensorFlow 2.2:
> https://github.com/boriswinner/RX580-rocM-tensorflow-ubuntu20.4-guide

Thank you @boriswinner, your guide is perfect, unlucky for me I haven't seen it before(I've seen all details in it, like kernel version downgrading when I installed first time Ubuntu-20.04.1 with **kernel5.8** + TensorFlow-Rocm I had wasted too much time see the issue reason)

I'll follow your guide soon, hope it'll work for me as well.

**Comment by ROCmSupport** [comment]
Thanks for reaching out.
gfx8 is not a supported config now.
We are not supporting gfx8 devices officially with ROCm and request you to follow our supported hardware section @ ROCm docs: [https://github.com/RadeonOpenCompute/ROCm#Hardware-and-Software-Support](url)
